torch/jit/_recursive.py

```diff
@@ -403,8 +384,58 @@ def init_fn(script_module):
         # nn.Module.forward)
         script_module.__dict__[name] = script_method
 
+    # In order to continue to expose module container functions to python,
+    # we add on the methods that we had previously exposed in the previous
+    # version of this api.
+    if isinstance(nn_module, (ModuleList, Sequential)):
```

Comment:
It would be awesome of we could eliminate any special handling for ModuleContainers here at all. The generic mechanism we want here is "make sure this Python method on `nn_module` shows up on the `script_module`". One way to do this it to have an internal decorator called like `@_copy_to_script` (we need a better name), which specifies that a given method should be copied to the new Python ScriptModule. These decorators can go on all the container modules, and `ScriptModule` creation can just process the decorators and do the copy without encoding special-case policies.

Reply:
These are special-cased methods that exist only so that we continue to expose the same api for these built-in containers as we did previously. I don't know if it makes sense to generalize it. Another possibility we can at least consider is to break the api for the constant modules. We'd only be breaking it for their usage as scripted modules within python, not within torchscript.